### ISSUE TITLE
Result modals do not close on ESC

### ISSUE THREAD

**Issue by emmastephenson** [CDCgov/prime-simplereport#4085]
## Area or Component
Results page modals:
- Download results
- Print results
- Correct results
- Text/email results
- View details

## Quick Summary
The modals should close on hitting the `esc` key.

### How to Replicate
1. Open a modal on the results page (like "Download results")
2. Hit esc
3. See that nothing happens

## Accessibility Violation
ESC key does not close modal: https://cdc-devtools.dequecloud.com/rule-help/en/modal-no-esc

## Action Requested
The modal should close on hitting the ESC key. #4028 has additional implementation details.

## Original Deque Issue
https://docs.google.com/spreadsheets/d/1b6xjubAN5Uw11DV4_B5zvSu7HeDI6Z-DpV4DMcToqAY/edit#gid=1978534823&range=17:18
